lib/matplotlib/widgets.py

```diff
@@ -1487,7 +1488,10 @@ def _press(self, event):
         self.rect.set_visible(self.visible)
         if self.span_stays:
             self.stay_rect.set_visible(False)
-
+            # really force a draw so that the stay rect is not in
+            # the blit background
+            if self.useblit:
+                self.canvas.draw()
```

Comment:
but why is this necessary? Is this essentially forcing a remake of the background buffer? Should these events be responsible for triggering direct draw calls? Why can't this be deferred to later?